File: python/programmers/level2/스킬트리.py
def solution(skill, skill_trees):
    answer = len(skill_trees)
    for skill_tree in skill_trees:
        temp = skill
        for alpha in skill_tree:
            if alpha in skill:
                if alpha != temp[0]:
                    answer -= 1
                    break
                temp = temp[1:]
            
    return answer

File: 프로그래머스/lv2/49993. 스킬트리/스킬트리.java
class Solution {
    public int solution(String skill, String[] skill_trees) {
        int answer = skill_trees.length;
        for(String skillTree :skill_trees){
            int sI = 0;
            for(String st : skillTree.split("")){
                if(skill.indexOf(st) == -1){
                    continue;
                }else if(skill.indexOf(st) == sI){
                    sI++;
                }else{
                    answer--;
                    break;
                }
            }
        }
        
        
        return answer;
    }
}